### PR TITLE
Added alphabet string function

### PR DESCRIPTION
**Pull request by Deconkle**
Adds a E2 function that returns the alphabet as a string, all lowercase.

### CODE REVIEW

File: lua/entities/gmod_wire_expression2/core/custom/cfc_string.lua [hunk 1]
```diff
@@ -38,3 +38,6 @@ e2function number string:isLower()
     if this == string.lower( this ) then return 1 end
     return 0
 end
+e2function string alphabet()
+	return "abcdefghijklmnopqrstuvwxyz"
```
Comment (OverlordBleck) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_string.lua]:
```suggestion
    return "abcdefghijklmnopqrstuvwxyz"
```
Using space indentation

File: lua/entities/gmod_wire_expression2/core/custom/cl_cfc_e2_lib.lua [hunk 2]
```diff
@@ -28,6 +28,7 @@ E2Helper.Descriptions["hasNumber(s:)"]         = "Returns 1 if string contains n
 E2Helper.Descriptions["hasLetter(s:)"]         = "Returns 1 if string has letters, 0 otherwise."
 E2Helper.Descriptions["isUpper(s:)"]           = "Returns 1 if string is uppercase, 0 otherwise."
 E2Helper.Descriptions["isLower(s:)"]           = "Returns 1 if string is lowercase, 0 otherwise."
+E2Helper.Descriptions["alphabet()"]			   = "Returns the alphabet (a-z) as a lowercase string"
```
Comment (OverlordBleck) [inline lua/entities/gmod_wire_expression2/core/custom/cl_cfc_e2_lib.lua]: Gotta fix this one for spaces too

**Comment by PLally** [comment]
what do you think about adding a constant `E2Lib.registerConstant("ALPHABET", "abcd..")`?

**Comment by OverlordBleck** [comment]
Couldn't hurt sounds like a good idea to me.

**Comment by brandonsturgeon** [comment]
What if you want to modify the given string?

`alphabet().replace("%W", "1")` works but would `ALPHABET.replace("%W", "1")` ?

Or `alphabet().toUppercase()` etc. 

Will any in-place modifiers work without copying the value?

**Comment by brandonsturgeon** [comment]
That's why I suggested making a method for it, but if you can find an easy way to make a constant work it might be a better option

**Comment by PLally** [comment]
i was thinking both, like e2 has pi() and _PI.

**Comment by brandonsturgeon** [comment]
retest this please

File: lua/entities/gmod_wire_expression2/core/custom/cfc_string.lua [hunk 3]
```diff
@@ -38,3 +38,6 @@ e2function number string:isLower()
     if this == string.lower( this ) then return 1 end
     return 0
 end
+e2function string alphabet()
```
Comment (PLally) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_string.lua]: newline before this

**Comment by PLally** [comment]
https://github.com/CFC-Servers/cfc_e2_lib/pull/57/files

**Pull request closed without merging**